MapboxNavigation/NavigationViewLayout.swift

```diff
@@ -38,9 +38,6 @@ extension NavigationView {
     }
     
     public func reinstallRequiredConstraints() {
-        if let bottomBannerContainerHeightConstraint = bottomBannerContainerHeightConstraint {
```

Comment:
I'm not sure this fix allows to resolve current issue. Even though you remove `bottomBannerContainerHeightConstraint` default `height` values are still used (can be checked when rotating device).

I think good idea would be to create two projects:
1. CocoaPods based project which uses Navigation SDK `0.40.0` which uses `CustomBottomBarViewController. You can then try to change `bannerView` height there by calling:
```swift
bottomBanner.bannerView.heightAnchor.constraint(equalToConstant: 10.0).isActive = true
```

2. CocoaPods based project which uses Navigation SDK `1.0.0` which uses `CustomBottomBarViewController`. You can apply the same change in height and see if it's was applied correctly (as it's stated in original issue).

Next step I think would be to debug how fix in https://github.com/mapbox/mapbox-navigation-ios/pull/2613 breaks the ability to control bottom bar height: try to remove `reinstallRequiredConstraints` completely and apply new fix without hardcoding default height values for bottom bar, but at the same time without causing regression in https://github.com/mapbox/mapbox-navigation-ios/issues/2561. 

It might also make sense to create test sample which uses only `MBLMapView` with visible `logoView` and `attributionButton` and plain UIView below them to understand how layout behaves when those `MBLMapView` views are constrained to bottom view.

Reply:
I also remember that when applying fix for #2613 I thought that it'd be enough to set https://github.com/mapbox/mapbox-navigation-ios/blob/f95bab2a5fabdf55a05e960fdbf5e01ff2c6bca1/MapboxNavigation/NavigationViewLayout.swift#L10-L13 to fix the issue, but bottom bar height started increasing simultaneously. I think good idea would be to check possible reasons of this as well.

Reply:
Yes. The bottom banner view hight could be adjusted in both Navigation SDK `0.40.0` and `1.0.0` by the 
`bottomBanner.bannerView.heightAnchor.constraint(equalToConstant: 10.0).isActive = true`
But it will cause the overlapping of the `logoView` and the `attributionButton`, and caused the break of the above constraint. 

Because the `CustomBottomBannerView` is inside the `MBBannerContainerView`. The top and bottom of the `MBBannerContainerView` is AutoLayout set with the imageView, button and the navigationView. So it's safer the change the `CustomBottomBannerView` height by setting top and bottom anchor responding to the `MBBannerContainerView`, which is
 `bannerView.topAnchor.constraint(equalTo: view.topAnchor, constant: 10.0)).isActive = true` 
and the 
`bannerView.bottomAnchor.constraint(equalTo: safeArea.bottomAnchor, constant: -10.0).isActive = true` 
so whole `MBBannerContainerView` could still  follow the view height change in the `reinstallRequiredConstraints`. And the the `CustomBottomBannerView` only change it's height inside the `MBBannerContainerView`. It would not cause the problem in the `logoView` or the `attributionButton` overlapping either.

Reply:
Yes, the nextstep is to rewrite the `reinstallRequiredConstraints` , to replace the height constant with the auto layout constraints. So the user could edit the `superview.height` in `CustomBottomBannerView`, to adjust the whole `BottomBannerContainerView`'s height